torchgeo/datasets/millionaid.py

```diff
@@ -0,0 +1,397 @@
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+        assert task in self.tasks
+        assert split in self.splits
+        self.task = task
+        self.split = split
+
+        self._verify()
+
+        self.files = self._load_files(self.root)
+
+        self.classes = sorted(set(cls for f in self.files for cls in f["label"]))
+        self.class_to_idx: Dict[str, int] = {c: i for i, c in enumerate(self.classes)}
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        files = self.files[index]
+        image = self._load_image(files["image"])
+        cls_label = [self.class_to_idx[label] for label in files["label"]]
+        label = torch.tensor(cls_label, dtype=torch.long)
+        sample = {"image": image, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def _load_files(self, root: str) -> List[Dict[str, Any]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, and list of labels
+        """
+        imgs_no_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "*.jpg"))
+        )
+
+        imgs_subcat = list(
+            glob.glob(os.path.join(root, self.split, "**", "**", "**", "*.jpg"))
+        )
+
+        scenes = [p.split(os.sep)[-3] for p in imgs_no_subcat] + [
+            p.split(os.sep)[-4] for p in imgs_subcat
+        ]
+
+        subcategories = ["Missing" for p in imgs_no_subcat] + [
+            p.split(os.sep)[-3] for p in imgs_subcat
+        ]
+
+        classes = [p.split(os.sep)[-2] for p in imgs_no_subcat] + [
+            p.split(os.sep)[-2] for p in imgs_subcat
+        ]
+
+        if self.task == "multi-label":
+            labels = [
+                [sc, sub, c] if sub != "Missing" else [sc, c]
+                for sc, sub, c in zip(scenes, subcategories, classes)
+            ]
+        else:
+            labels = [[c] for c in classes]
+
+        images = imgs_no_subcat + imgs_subcat
+
+        files = [dict(image=img, label=l) for img, l in zip(images, labels)]
```

Comment:
`pyupgrade` (a new CI step we added recently) will probably complain about dict comprehension here. You can run `pip install pyupgrade` and `pyupgrade --py37-plus torchgeo/datasets/millionaid.py` to fix this.